### ISSUE TITLE
A runtime crash bug during analysis

### ISSUE THREAD

**Issue by kitty-1998** [soot-oss/SootUp#911]
Hi, I used SootUp to constrcut CHA call graph and there is a crash in the minimized code example listed below.

#### Input Code
```java
package edu.example;
import java.io.FileInputStream;
class A {
    public static A[] bar(int len) throws Exception {
        FileInputStream fis = new FileInputStream("PATH");
        return new A[len];
    }
}
class B {
    void foo(int len) {
        A[] arr;
        try {
            arr = A.bar(len);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
        for (A a : arr) {}
    }
}
```

My version: develop-5af272dcda

#### SootUp Code
```java
AnalysisInputLocation javaBase = new JavaClassPathAnalysisInputLocation(Path/to/JAVABASE, SourceType.Library);
AnalysisInputLocation classInput = new JavaClassPathAnalysisInputLocation(Path/to/input, SourceType.Application);
ArrayList<AnalysisInputLocation> allInput=new ArrayList<>();
allInput.add(classInput);
allInput.add(javaBase);
JavaView view = new JavaView(allInput);
CallGraphAlgorithm cha = new ClassHierarchyAnalysisAlgorithm(view);
CallGraph cg = cha.initialize(entryMethods);  // entry method is <edu.example.B: void foo(int)>
```

#### Stacktrace
```
java.lang.IllegalStateException: Failed to apply sootup.java.core.interceptors.Aggregator@6f1401f6 to <edu.example.B: void foo(int)>

	at sootup.java.bytecode.frontend.AsmMethodSource.resolveBody(AsmMethodSource.java:240)
	at sootup.core.model.SootMethod.lazyBodyInitializer(SootMethod.java:98)
	at com.google.common.base.Suppliers$NonSerializableMemoizingSupplier.get(Suppliers.java:186)
	at sootup.core.model.SootMethod.getBody(SootMethod.java:177)
	at sootup.callgraph.AbstractCallGraphAlgorithm.resolveAllCallsFromSourceMethod(AbstractCallGraphAlgorithm.java:218)
	at sootup.callgraph.AbstractCallGraphAlgorithm.processWorkList(AbstractCallGraphAlgorithm.java:176)
	at sootup.callgraph.AbstractCallGraphAlgorithm.constructCompleteCallGraph(AbstractCallGraphAlgorithm.java:85)
	at sootup.callgraph.ClassHierarchyAnalysisAlgorithm.initialize(ClassHierarchyAnalysisAlgorithm.java:65)
```

**Comment by JonasKlauke** [comment]
I assume the aggregator has a problem with the empty for loop. @swissiety you are the expert for intercepters 😄

**Comment by swissiety** [comment]
Thanks for the report - we have a missing exceptional flow for `arr = A.bar(len);` which leads to a lonely island of stmts - representing the traphandler - that are not integrated into the flow - see [GraphViz]( http://magjac.com/graphviz-visual-editor/?dot=digraph+G+%7B%0A%09compound%3Dtrue%0A%09labelloc%3Db%0A%09style%3Dfilled%0A%09color%3Dgray90%0A%09node+%5Bshape%3Dbox%2Cstyle%3Dfilled%2Ccolor%3Dwhite%5D%0A%09edge+%5Bfontsize%3D10%2Carrowsize%3D1.5%2Cfontcolor%3Dgrey40%5D%0A%09fontsize%3D10%0A%0A%2F%2F++lines+%5B13%3A+13%5D+%0A%09subgraph+cluster_1754894440+%7B+%0A%09%09label+%3D+%22Block+%231%22%0A%09%091888442711%5Blabel%3D%22this+%3A%3D+%40this%3A+B%22%2Cshape%3DMdiamond%2Ccolor%3Dgrey50%2Cfillcolor%3Dwhite%5D%0A%09%091998767043%5Blabel%3D%22l1+%3A%3D+%40parameter0%3A+int%22%5D%0A%0A%09%091888442711+-%3E+1998767043%0A%09%7D%0A%091998767043%3As+-%3E+154482552%3An%0A%0A%2F%2F++lines+%5B16%3A+16%5D+%0A%09subgraph+cluster_514455215+%7B+%0A%09%09label+%3D+%22Block+%232%22%0A%09%09154482552%5Blabel%3D%22goto%22%5D%0A%09%7D%0A%09154482552%3As+-%3E+944427387%3An%0A%0A%2F%2F++lines+%5B17%3A+17%5D+%0A%09subgraph+cluster_1360657223+%7B+%0A%09%09label+%3D+%22Block+%233%22%0A%09%09944427387%5Blabel%3D%22l3+%3D+staticinvoke+%26lt%3BA%3A+A%5B%5D+bar%28int%29%26gt%3B%28l1%29%22%5D%0A%09%091997859171%5Blabel%3D%22l4+%3D+lengthof+l3%22%5D%0A%09%09550402284%5Blabel%3D%22l5+%3D+0%22%5D%0A%0A%09%09944427387+-%3E+1997859171+-%3E+550402284%0A%09%7D%0A%09550402284%3As+-%3E+365181913%3An%0A%0A%2F%2F++lines+%5B17%3A+17%5D+%0A%09subgraph+cluster_1912962767+%7B+%0A%09%09label+%3D+%22Block+%234%22%0A%09%09365181913%5Blabel%3D%22if+l5+%26gt%3B%3D+l4%22%5D%0A%09%7D%0A%09365181913%3As+-%3E+959869407%3An%5Blabel%3D%22false%22%5D%0A%09365181913%3As+-%3E+116237769%3An%5Blabel%3D%22true%22%5D%0A%0A%2F%2F++lines+%5B17%3A+17%5D+%0A%09subgraph+cluster_1449263511+%7B+%0A%09%09label+%3D+%22Block+%235%22%0A%09%09959869407%5Blabel%3D%22l6+%3D+l3%5Bl5%5D%22%5D%0A%09%091594199808%5Blabel%3D%22l5+%3D+l5+%2B+1%22%5D%0A%09%091984990929%5Blabel%3D%22goto%22%5D%0A%0A%09%09959869407+-%3E+1594199808+-%3E+1984990929%0A%09%7D%0A%091984990929%3Ae+-%3E+365181913%3An%0A%0A%2F%2F++lines+%5B18%3A+18%5D+%0A%09subgraph+cluster_1438098656+%7B+%0A%09%09label+%3D+%22Block+%236%22%0A%09%09116237769%5Blabel%3D%22return%22%2Cshape%3DMdiamond%2Ccolor%3Dgrey50%2Cfillcolor%3Dwhite%5D%0A%09%7D%0A%0A%2F%2F++lines+%5B-1%3A+15%5D+%0A%09subgraph+cluster_411506101+%7B+%0A%09%09label+%3D+%22Block+%237%22%0A%09%091637290981%5Blabel%3D%22%24stack7+%3A%3D+%40caughtexception%22%5D%0A%09%091905485420%5Blabel%3D%22l3+%3D+%24stack7%22%5D%0A%09%09551479935%5Blabel%3D%22%24stack8+%3D+new+java.lang.RuntimeException%22%5D%0A%09%0958940486%5Blabel%3D%22specialinvoke+%24stack8.%26lt%3Bjava.lang.RuntimeException%3A+void+%26lt%3Binit%26gt%3B%28java.lang.Throwable%29%26gt%3B%28l3%29%22%5D%0A%09%09501107890%5Blabel%3D%22throw+%24stack8%22%2Cshape%3DMdiamond%2Ccolor%3Dgrey50%2Cfillcolor%3Dwhite%5D%0A%0A%09%091637290981+-%3E+1905485420+-%3E+551479935+-%3E+58940486+-%3E+501107890%0A%09%7D%0A%0A%7D)

**Comment by swissiety** [comment]
should be fixed by #912
